lib/modules/watcher/index.js

```diff
@@ -0,0 +1,201 @@
+      filesToWatch,
+      function (eventName, path) {
+        self.logger.info(`${eventName}: ${path}`);
+        self.events.emit('file-' + eventName, 'asset', path);
+        self.events.emit('file-event', 'asset', path);
+      },
+      function () {
+        callback();
+      }
+    );
+  }
+
+  watchContracts(embarkConfig, callback) {
+    let self = this;
+    this.watchFiles(
+      [embarkConfig.contracts],
+      function (eventName, path) {
+        self.logger.info(`${eventName}: ${path}`);
+        self.events.emit('file-' + eventName, 'contract', path);
+        self.events.emit('file-event', 'contract', path);
+      },
+      function () {
+        callback();
+      }
+    );
+  }
+
+  watchWebserverConfig(embarkConfig, callback) {
+    let self = this;
+    let webserverConfig;
+    if (typeof embarkConfig.config === 'object') {
+      if (!embarkConfig.config.webserver) {
+        return;
+      }
+      webserverConfig = embarkConfig.config.webserver;
+    } else {
+      let contractsFolder = embarkConfig.config.replace(/\\/g, '/');
+      if (contractsFolder.charAt(contractsFolder.length - 1) !== '/') {
+        contractsFolder += '/';
+      }
+      webserverConfig = [`${contractsFolder}**/webserver.json`, `${contractsFolder}**/webserver.js`];
+    }
+    this.watchFiles(webserverConfig,
+      function (eventName, path) {
+        self.logger.info(`${eventName}: ${path}`);
+        self.events.emit('webserver:config:change', 'config', path);
+      },
+      function () {
+        callback();
+      }
+    );
+  }
+
+  watchContractConfig(embarkConfig, callback) {
+    let self = this;
+    let contractConfig;
+    if (typeof embarkConfig.config === 'object' || embarkConfig.config.contracts) {
+      contractConfig = embarkConfig.config.contracts;
+    } else {
+      let contractsFolder = embarkConfig.config.replace(/\\/g, '/');
+      if (contractsFolder.charAt(contractsFolder.length - 1) !== '/') {
+        contractsFolder += '/';
+      }
+      contractConfig = [`${contractsFolder}**/contracts.json`, `${contractsFolder}**/contracts.js`];
+    }
+    this.watchFiles(contractConfig,
+      function (eventName, path) {
+        self.logger.info(`${eventName}: ${path}`);
+        self.events.emit('file-' + eventName, 'config', path);
+        self.events.emit('file-event', 'config', path);
+      },
+      function () {
+        callback();
+      }
+    );
+  }
+
+  watchPipelineConfig(embarkConfig, callback) {
+    const dappPipelineConfigPath = fs.dappPath(embarkConfig.config, DAPP_PIPELINE_CONFIG_FILE);
+    const dappWebpackConfigPath = fs.dappPath(embarkConfig.config, `../${DAPP_WEBPACK_CONFIG_FILE}`);
+
+    this.watchFiles([
+      dappPipelineConfigPath,
+      dappWebpackConfigPath
+    ], (eventName, path) => {
+      this.logger.info(`${eventName}: ${path}`);
+      this.events.emit('file-' + eventName, 'config', path);
+      this.events.emit('file-event', 'config', path);
+    }, callback)
```

Comment:
This is very similar to other `watch*()` methods inside this class. I duplicated this for consistency, but we might want to reconsider introducing multiple watch methods with partially the same algorithm. 

Maybe we can make `Watcher` collect all file paths once and then call `watchFiles()` once for all of them instead of redoing this in several places.